File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        
        if(x<=Integer.MIN_VALUE || x>=Integer.MAX_VALUE) return 0;
        
        double rev=0;
        int orginal = x;

        while(x!=0) {
            if(orginal>0) {
                int a = x%10;
                rev = rev*10 + a;
                x = x/10; 
                if(rev>=Integer.MAX_VALUE) {
                    rev = 0;
                    break;
                }
            } 
            else {
                x = Math.abs(x);
                int a = x%10;
                rev = rev*10 - a;
                x = x/10;
                if(rev<=Integer.MIN_VALUE) {
                    rev = 0;
                    break;
                }
            }
        }

        int result = (int)rev;
        return result;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        
        sx = str(x)
        nsx = ""
        flag = True
        if x < 0:
            flag = False

        for c in sx:
            if c.isdigit():
                nsx += c
            
        res = int(nsx[::-1]) if flag else int(nsx[::-1]) * -1

        if (-2 ** 31) <= res < ((2 ** 31) - 1):
            return res
        else:
            return 0